app/src/main/java/pm/gnosis/heimdall/ui/tokens/info/TokenInfoActivity.kt

```diff
@@ -0,0 +1,120 @@
+import android.support.v7.app.AlertDialog
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.rxkotlin.plusAssign
+import io.reactivex.subjects.PublishSubject
+import kotlinx.android.synthetic.main.layout_token_info.*
+import pm.gnosis.heimdall.HeimdallApplication
+import pm.gnosis.heimdall.R
+import pm.gnosis.heimdall.common.di.components.DaggerViewComponent
+import pm.gnosis.heimdall.common.di.modules.ViewModule
+import pm.gnosis.heimdall.common.utils.snackbar
+import pm.gnosis.heimdall.common.utils.subscribeForResult
+import pm.gnosis.heimdall.common.utils.toast
+import pm.gnosis.heimdall.data.repositories.models.ERC20Token
+import pm.gnosis.heimdall.ui.base.BaseActivity
+import pm.gnosis.utils.asEthereumAddressStringOrNull
+import pm.gnosis.utils.hexAsEthereumAddressOrNull
+import timber.log.Timber
+import java.math.BigInteger
+import javax.inject.Inject
+
+class TokenInfoActivity : BaseActivity() {
+    @Inject
+    lateinit var viewModel: TokenInfoContract
+
+    private lateinit var tokenAddress: BigInteger
+    private val removeClicksSubject = PublishSubject.create<Unit>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        inject()
+
+        val addressExtra = intent.extras.getString(ADDRESS_EXTRA, "").hexAsEthereumAddressOrNull()
+        if (addressExtra != null) {
+            tokenAddress = addressExtra
+        } else {
+            toast(R.string.invalid_ethereum_address)
+            finish()
+        }
+
+        setContentView(R.layout.layout_token_info)
+        registerToolbar(layout_token_info_toolbar)
+        layout_token_info_toolbar.inflateMenu(R.menu.token_info_menu)
+        layout_token_info_toolbar.setOnMenuItemClickListener {
+            when (it.itemId) {
+                R.id.token_info_menu_delete -> showDeleteDialog()
+            }
+            true
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        disposables += viewModel.observeToken(tokenAddress)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(this::onTokenInfo, this::onTokenInfoError)
+
+        disposables += removeClicksSubject
+                .flatMap { viewModel.removeToken() }
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeForResult(this::onTokenDeleted, this::onTokenDeleteError)
+    }
+
+    private fun onTokenInfo(token: ERC20Token) {
+        layout_token_info_toolbar.title = token.name ?: getString(R.string.token_info)
+        layout_token_info_name.text = token.name ?: "-"
+        layout_token_info_symbol.text = token.symbol ?: "-"
+        layout_token_info_decimals.text = token.decimals.toString()
+        layout_token_info_address.text = token.address.asEthereumAddressStringOrNull() ?: "-"
+
+        val menuItem = layout_token_info_toolbar.menu.findItem(R.id.token_info_menu_delete)
+        menuItem.isVisible = !token.verified
```

Comment:
Should we not hide the delte button by default? Else it will be visible until the token info is loaded